src/services/chains.rs

```diff
@@ -0,0 +1,34 @@
+use crate::cache::cache_operations::RequestCached;
+use crate::config::{
+    base_config_service_url, chain_info_cache_duration, chain_info_request_timeout,
+};
+use crate::models::chains::ChainInfo;
+use crate::models::commons::Page;
+use crate::utils::context::Context;
+use crate::utils::errors::ApiResult;
+use reqwest::Url;
+use std::collections::HashMap;
+
+pub async fn get_chains_paginated(
+    context: &Context<'_>,
+    limit: &Option<String>,
+) -> ApiResult<Page<ChainInfo>> {
+    let mut queries: HashMap<&str, String> = HashMap::new();
+    if let Some(limit) = limit {
+        queries.insert("limit", limit.to_string());
+    }
```

Comment:
Like I mentioned above, maybe it is worth building some `structs` and methods that adjust the pagination from the config and core services to the gateway's. Totally out of the scope of this PR though :)